src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/SetAzureSqlDatabaseAuditingPolicy.cs

```diff
@@ -120,11 +120,20 @@ private void ApplyUserInputToBlobAuditingModel(DatabaseBlobAuditingPolicyModel m
             {
                 model.RetentionInDays = RetentionInDays;
             }
+
             if (StorageAccountName != null)
             {
                 model.StorageAccountName = StorageAccountName;
             }
 
+            if (!string.IsNullOrEmpty(StorageKeyType))
+            {
+                // the user enter a key type - we use it (and override the previously defined key type)
+                model.StorageKeyType = (StorageKeyType == SecurityConstants.Primary)
+                    ? StorageKeyKind.Primary
+                    : StorageKeyKind.Secondary;
+            }
+
             if (AuditActionGroup != null &&  AuditActionGroup.Length != 0)
```

Comment:
This is an array and not a String. I will finish handling the parameters in another task as agreed with Adam and Mark.